lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -181,12 +185,321 @@ def test_duplicate(self, selinux_disabled, server_config, make_logger, tarball):
             tree.create("ABC", source_tarball)
         assert exc.value.tarball == Dataset.stem(source_tarball)
 
-    def test_find(self, selinux_enabled, server_config, make_logger, tarball):
+    def test_unpack(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
+    ):
+        """Test to check the unpacking functionality of the FileTree"""
+        source_tarball, source_md5, md5 = tarball
+
+        def mock_run(args, **kwargs):
+            assert kwargs["capture_output"]
+            assert kwargs["text"]
+
+            tarball_name = source_tarball.name.removesuffix(".tar.xz")
+            tar_target = "--file=" + str(
+                server_config.ARCHIVE / "ABC" / source_tarball.name
+            )
+            assert args[0] == "find" or args[0] == "tar" and tar_target in args
+            assert (
+                kwargs["cwd"]
+                == server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
+            )
+
+            assert kwargs["stdin"] == subprocess.DEVNULL
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+            )
+
+        with monkeypatch.context() as m:
+            m.setattr(subprocess, "run", mock_run)
```

Comment:
We're missing a mock for `Path.mkdir()`!  (Ditto for all the other new test scenarios....)

Reply:
tree.create() is expecting that directory to be created else there are a lot nested issues.

Reply:
We want to get to the point where we have proper unit tests as well as nice functional tests, and we want to get away from the current methodology which creates an annoying mixture.  I would prefer that all new tests be explicitly one or the other (so, this test shouldn't be utilizing the local file system), otherwise we're not moving ourselves in the right direction.

That said, I realize that the realities of our situation.  So, to the extent that this test isn't going to fully mock the CUT's external dependencies, it should have a big, nasty-looking code comment pointing that out.  (I.e., it should have a `TODO` (or even a `FIXME`) tag.)  Otherwise, not only are we increasing our technical debt, we're sort of hiding it and making our future job harder.

So, if it's going to be too much of a problem to add the missing mock, please document the situation so that it is plain for the next developer who comes along.  (This is the problem of letting the "unit" get too big -- testing it becomes cumbersome -- which is why we prefer small, targeted functions which can be readily isolated for testing.)